src/shared/ThemeContext/ThemeContext.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { createContext, FC, ReactNode, useEffect, useState } from 'react'
+
+export enum Theme {
+  LIGHT = 'light',
+  DARK = 'dark',
+}
+
+interface ThemeContextProps {
+  theme: Theme
+  setTheme: (theme: Theme) => void
+}
+
+const defaultThemeContext: ThemeContextProps = {
+  theme: Theme.LIGHT,
+  setTheme: () => {},
+}
+
+export const ThemeContext =
+  createContext<ThemeContextProps>(defaultThemeContext)
+
+interface ThemeContextProviderProps {
+  children: ReactNode
+}
+
+export const ThemeContextProvider: FC<ThemeContextProviderProps> = ({
+  children,
+}) => {
+  const currentTheme = (localStorage.getItem('theme') as Theme) || Theme.LIGHT
+  const [theme, setTheme] = useState<Theme>(currentTheme)
+
+  useEffect(() => {
+    document.body.classList.remove(Theme.LIGHT, Theme.DARK)
+    document.body.classList.add(theme)
+
+    localStorage.setItem('theme', theme)
+  }, [theme])
+
+  const handleSetTheme = (newTheme: Theme) => {
+    setTheme(newTheme)
+  }
```

Comment:
Can you wrap this in a `useCallback` to memo'ize it, so there's no weird re-render issues that may pop up

Reply:
Ah good catch!
Actually we no longer need this helper at all anymore - just returned `setTheme` directly!